build-tools/github/validate-yaml

```diff
@@ -1,60 +1,47 @@
 #!/bin/bash
 set -eu
 scriptdir=$(cd $(dirname $0) && pwd)
-cd $scriptdir
+cd $scriptdir/../..
 
 echo "------> Validating YAML"
 
 # Validate adventure files against their schema
 #
+# We'll go into every directory that has a <xxx>.schema.json file,
+# and validate all YAML files in that same directory against the schema file.
+#
 # The error messages produced by pajv are pretty bad, but we will just
 # say that getting an error is better than getting NO error.
 #
 # If we ever care enough to do something about this, we should be interpreting
 # the JSON error outputs of this tool and re-interpreting them with useful
 # error messages.
 
-
 # For magical and mysterious reasons, on GitHub actions using 'npx pajv validate' (which
 # would normally be the way to run this command), hangs. It works fine on two of my machines,
 # and also in an Ubuntu Docker container, and 'npx pajv help' works as well... but
 # 'npx pajv validate' just hangs. Running the 'pajv' binary directly without the use of
 # 'npx' does work... so we're just going to ¯\_(ツ)_/¯ and do that.
 
-pajv=$scriptdir/../../node_modules/.bin/pajv
-
-echo "------> Validating adventures"
-cd ../../content/adventures/
-$pajv validate \
-    --errors=text \
-    -s "adventures.schema.json" \
-    -d "*.yaml"
-echo "------> Adventure yaml files are valid"
-
-echo "------> Validating achievements"
-cd ../../content/achievements/
-$pajv validate \
-    --errors=text \
-    -s "achievements.schema.json" \
-    -d "*.yaml"
-echo "------> Achievements yaml files are valid"
-
-echo "------> Validating quizzes"
-cd ../quizzes/
-$pajv validate \
-    --errors=text \
-    -s "quizzes.schema.json" \
-    -d "*.yaml"
-echo "------> Quiz yaml files are valid"
-
-echo "------> Validating parsons"
-cd ../parsons/
-$pajv validate \
-    --errors=text \
-    -s "parsons.schema.json" \
-    -d "*.yaml"
-echo "------> Parsons yaml files are valid"
-
-# Validate-texts is producing a lot of noise and it will never fail the build outright
-# anyway (if it would, pajv would have complained earlier), so might as well disable it for now.
-# node $scriptdir/validate-texts.js $scriptdir
+pajv=node_modules/.bin/pajv
+
+all_schemas=$(find content -name \*.schema.json)
+
+failures=false
+
+for schema in $all_schemas; do
+    dir=$(dirname $schema)
+    echo "------> Validating $(basename $dir)"
+
+    # Run pajv, remove all the lines that end in the word ' valid' -- those are not interesting
```

Comment:
ah nice!! That long list of output was annoying but I had not thought of a way to get rid of it!